azurerm/resource_arm_public_ip_prefix.go

```diff
@@ -0,0 +1,191 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-08-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPublicIpPrefix() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPublicIpPrefixCreateUpdate,
+		Read:   resourceArmPublicIpPrefixRead,
+		Update: resourceArmPublicIpPrefixCreateUpdate,
+		Delete: resourceArmPublicIpPrefixDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
I'm not sure if this is required? The read function should be doing the same check on the `id` passed into it?

Reply:
I have to admit this was a case of copy and paste from the azurerm_public_ip resource and I wasn't entirely sure why it was there.

Removed now